pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -604,22 +646,41 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 
 	if provResult.Dirty {
 		info.host.ClearError()
-		return actionContinue{provResult.RequeueAfter}
+		return actionContinue{delay: provResult.RequeueAfter, dirty: true}
+	}
+
+	desiredPowerOnState := info.host.Spec.Online
+
+	if hasRebootAnnotation(info.host) {
```

Comment:
We originally said that we would only pay attention to reboot requests when the server was provisioned, so I think we should also check that we are being called from the Provisioned state (and not the Ready/Available state).
Now that I think about it, this was related to the whole LastBootedAt thing - that enabled us to ensure we didn't bother to reboot a just-provisioned server due to a reboot annotation that predated the provisioning. I think we can handle that by just clearing the annotations again at the start of provisioning (just like we do at the end of deprovisioning).

Reply:
Do you want me to check that the host is in `ExternallyProvisioned || Provisioned` state or add additional argument to `manageHostPower()` ?

Reply:
@zaneb for now I implemented the first option